### PR TITLE
Move Pull Request builder into scripts

### PR DESCRIPTION
**Pull request by Sanchiz**
## Job configuration
image

## Change record
- Moved shell commands into [cibox-code-deploy](https://github.com/Sanchiz/cibox/tree/builder_job/core/cibox-project-builder/files/vagrant/box/cibox/custom/cibox-code-deploy) role
- Trigger [deploy_code.yml](https://github.com/Sanchiz/cibox/blob/builder_job/core/cibox-project-builder/files/vagrant/box/cibox/jobs/deploy_code.yml) and run that role at the beginning of PR builde
- Got rid of shell scripts in PR_BUILDER

### CODE REVIEW

File: core/cibox-jenkins/templates/jobs/pr_builder.xml.j2 [hunk 1]
```diff
@@ -1,30 +1,30 @@
...
+    </com.coravy.hudson.plugins.github.GithubProjectProperty>
+  </properties>
+  <scm class="hudson.plugins.git.GitSCM" plugin="git@2.2.5">
+    <configVersion>2</configVersion>
+    <userRemoteConfigs>
+      <hudson.plugins.git.UserRemoteConfig>
+        <name>origin</name>
+        <refspec>+refs/pull/*:refs/remotes/origin/pr/*</refspec>
+        <url>{{ github_url }}.git</url>
+        <credentialsId>c55a496e-7717-4f51-88d5-1a50b478ba55</credentialsId>
```
Comment (podarok) [inline core/cibox-jenkins/templates/jobs/pr_builder.xml.j2]: Let's follow-up handling Id via ansible template variable at some future point

File: core/cibox-jenkins/templates/jobs/pr_builder.xml.j2 [hunk 2]
```diff
@@ -39,73 +39,271 @@
         <timeout>5</timeout>
       </hudson.plugins.git.extensions.impl.SubmoduleOption>
       <hudson.plugins.git.extensions.impl.RelativeTargetDirectory>
-        <relativeTargetDir>new_pull_request</relativeTargetDir>
+        <relativeTargetDir>sourcecode</relativeTargetDir>
```
Comment (podarok) [inline core/cibox-jenkins/templates/jobs/pr_builder.xml.j2]: Can we make it configurable?

File: core/cibox-jenkins/templates/jobs/pr_builder.xml.j2 (hunk 1 above)
Comment (Sanchiz) [inline core/cibox-jenkins/templates/jobs/pr_builder.xml.j2]: It appears not only here. That's good catch for follow up.

File: core/cibox-jenkins/templates/jobs/pr_builder.xml.j2 [hunk 3]
```diff
@@ -39,73 +39,271 @@
...
-            <permitAll>false</permitAll>
-            <commentFilePath>${WORKSPACE}/commentinfo.md</commentFilePath>
-            <whitelist></whitelist>
-            <autoCloseFailedPullRequests>false</autoCloseFailedPullRequests>
-            <whiteListTargetBranches>
-                <org.jenkinsci.plugins.ghprb.GhprbBranch>
-                    <branch></branch>
-                </org.jenkinsci.plugins.ghprb.GhprbBranch>
-            </whiteListTargetBranches>
-            <project>SRS_PR_BUILDER</project>
```
Comment (podarok) [inline core/cibox-jenkins/templates/jobs/pr_builder.xml.j2]: Let's get rid of internal namings here

File: core/cibox-jenkins/templates/jobs/pr_builder.xml.j2 [hunk 4]
```diff
@@ -39,73 +39,271 @@
...
-    </buildWrappers>
-</project>
+  </scm>
+  <canRoam>true</canRoam>
+  <disabled>false</disabled>
+  <blockBuildWhenDownstreamBuilding>false</blockBuildWhenDownstreamBuilding>
+  <blockBuildWhenUpstreamBuilding>false</blockBuildWhenUpstreamBuilding>
+  <triggers>
+    <org.jenkinsci.plugins.ghprb.GhprbTrigger plugin="ghprb@1.15-0">
+      <spec>* * * * *</spec>
```
Comment (podarok) [inline core/cibox-jenkins/templates/jobs/pr_builder.xml.j2]: Every minute? Are you sure?
Let's make it at least every 3 minutes

File: core/cibox-jenkins/templates/jobs/pr_builder.xml.j2 [hunk 5]
```diff
@@ -39,73 +39,271 @@
...
+      <adminlist></adminlist>
+      <allowMembersOfWhitelistedOrgsAsAdmin>true</allowMembersOfWhitelistedOrgsAsAdmin>
+      <orgslist>propeoplemd</orgslist>
+      <cron>* * * * *</cron>
```
Comment (podarok) [inline core/cibox-jenkins/templates/jobs/pr_builder.xml.j2]: Every minute? Are you sure?
Let's make it at least every 3 minutes

File: core/cibox-jenkins/templates/jobs/pr_builder.xml.j2 [hunk 6]
```diff
@@ -39,73 +39,271 @@
...
+      <permitAll>false</permitAll>
+      <commentFilePath>${WORKSPACE}/commentinfo.md</commentFilePath>
+      <whitelist></whitelist>
+      <autoCloseFailedPullRequests>false</autoCloseFailedPullRequests>
+      <whiteListTargetBranches>
+        <org.jenkinsci.plugins.ghprb.GhprbBranch>
+          <branch></branch>
+        </org.jenkinsci.plugins.ghprb.GhprbBranch>
+      </whiteListTargetBranches>
+      <project>PR_BUILDER</project>
```
Comment (podarok) [inline core/cibox-jenkins/templates/jobs/pr_builder.xml.j2]: Should be configurable, depending from the name of created project builder

File: core/cibox-jenkins/templates/jobs/pr_builder.xml.j2 [hunk 7]
```diff
@@ -39,73 +39,271 @@
...
+      <limit></limit>
+      <tags></tags>
+      <skippedTags></skippedTags>
+      <startAtTask></startAtTask>
+      <credentialsId></credentialsId>
+      <sudo>false</sudo>
+      <sudoUser></sudoUser>
+      <forks>5</forks>
+      <unbufferedOutput>true</unbufferedOutput>
+      <colorizedOutput>false</colorizedOutput>
```
Comment (podarok) [inline core/cibox-jenkins/templates/jobs/pr_builder.xml.j2]: true

File: core/cibox-jenkins/templates/jobs/pr_builder.xml.j2 [hunk 8]
```diff
@@ -39,73 +39,271 @@
...
```
Comment (podarok) [inline core/cibox-jenkins/templates/jobs/pr_builder.xml.j2]: true

File: core/cibox-jenkins/templates/jobs/pr_builder.xml.j2 [hunk 9]
```diff
@@ -39,73 +39,271 @@
...
```
Comment (podarok) [inline core/cibox-jenkins/templates/jobs/pr_builder.xml.j2]: true

File: core/cibox-jenkins/templates/jobs/pr_builder.xml.j2 [hunk 10]
```diff
@@ -39,73 +39,271 @@
...
+      <copyCredentialsInWorkspace>false</copyCredentialsInWorkspace>
+      <extraVars>
+        <org.jenkinsci.plugins.ansible.ExtraVar>
+          <key>php_env_vars</key>
+          <value>APP_ENV=dev</value>
+          <hidden>false</hidden>
+        </org.jenkinsci.plugins.ansible.ExtraVar>
+        <org.jenkinsci.plugins.ansible.ExtraVar>
+          <key>mysql_user</key>
+          <value>root</value>
```
Comment (podarok) [inline core/cibox-jenkins/templates/jobs/pr_builder.xml.j2]: can we grab this via variables?

File: core/cibox-jenkins/templates/jobs/pr_builder.xml.j2 [hunk 11]
```diff
@@ -39,73 +39,271 @@
...
+          <hidden>false</hidden>
+        </org.jenkinsci.plugins.ansible.ExtraVar>
+        <org.jenkinsci.plugins.ansible.ExtraVar>
+          <key>mysql_password</key>
+          <value>root</value>
```
Comment (podarok) [inline core/cibox-jenkins/templates/jobs/pr_builder.xml.j2]: can we grab this via variables?

File: core/cibox-project-builder/files/vagrant/box/cibox/jobs/build.yml [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+    workspace: ./
+    build_number: 1
+    server_docroot_folder: /var/www
+    project_docroot_folder: docroot
+    php_env_vars: 'APP_ENV=dev'
+    mysql_user: 'root'
+    mysql_password: 'root'
+    mysql_db: drupal{{ build_number }}
+    drupal_folder: "{{ server_docroot_folder }}/build{{ build_number }}"
+    site_url: "http://10.0.2.15/build{{ build_number }}"
```
Comment (podarok) [inline core/cibox-project-builder/files/vagrant/box/cibox/jobs/build.yml]: Why the IP is hardcoded? Could we grab it from server somehow and make it dynamic?

File: core/cibox-project-builder/files/vagrant/box/cibox/jobs/build.yml [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+  connection: local
+  gather_facts: yes
+
+  vars:
+    workspace: ./
+    build_number: 1
+    server_docroot_folder: /var/www
+    project_docroot_folder: docroot
+    php_env_vars: 'APP_ENV=dev'
+    mysql_user: 'root'
```
Comment (podarok) [inline core/cibox-project-builder/files/vagrant/box/cibox/jobs/build.yml]: These needs to be global vars, I assume

File: core/cibox-jenkins/templates/jobs/pr_builder.xml.j2 (hunk 6 above)
Comment (Sanchiz) [inline core/cibox-jenkins/templates/jobs/pr_builder.xml.j2]: This is out of current PR. Should be as follow up.

File: core/cibox-jenkins/templates/jobs/pr_builder.xml.j2 (hunk 7 above)
Comment (Sanchiz) [inline core/cibox-jenkins/templates/jobs/pr_builder.xml.j2]: It doesn't work as expected.

File: core/cibox-jenkins/templates/jobs/pr_builder.xml.j2 (hunk 8 above)
Comment (Sanchiz) [inline core/cibox-jenkins/templates/jobs/pr_builder.xml.j2]: It doesn't work as expected.

File: core/cibox-jenkins/templates/jobs/pr_builder.xml.j2 (hunk 9 above)
Comment (Sanchiz) [inline core/cibox-jenkins/templates/jobs/pr_builder.xml.j2]: It doesn't work as expected.

File: core/cibox-jenkins/templates/jobs/pr_builder.xml.j2 (hunk 10 above)
Comment (Sanchiz) [inline core/cibox-jenkins/templates/jobs/pr_builder.xml.j2]: Where should live that variable?

File: core/cibox-jenkins/templates/jobs/pr_builder.xml.j2 (hunk 11 above)
Comment (Sanchiz) [inline core/cibox-jenkins/templates/jobs/pr_builder.xml.j2]: Where should live that variable?

**Comment by podarok** [comment]
ping

**Pull request merged**